pandas/tests/indexes/numeric/test_numeric.py

```diff
@@ -327,18 +324,16 @@ def test_identical(self, simple_index, dtype):
         assert not index.astype(dtype=object).identical(index.astype(dtype=dtype))
 
     def test_cant_or_shouldnt_cast(self):
-        msg = (
-            "String dtype not supported, "
-            "you may need to explicitly cast to a numeric type"
-        )
+        msg = f"data is not compatible with {self._index_cls.__name__}"
+
         # can't
         data = ["foo", "bar", "baz"]
-        with pytest.raises(TypeError, match=msg):
+        with pytest.raises(ValueError, match=msg):
             self._index_cls(data)
 
         # shouldn't
         data = ["0", "1", "2"]
-        with pytest.raises(TypeError, match=msg):
```

Comment:
probably requires a whatsnew note

Reply:
or could just keep the string_error_whatever to keep this tightly focused and avoid the API change

Reply:
This error can in pandas <2.0 only be hit by instantiating `Int64Index` et al. directly, because `Index` will give a object dtype to this. `Int64Index` will be removed in pandas 2.0, so IMO a whatsnew notes should not be added?

Doesn't hit this either it we combine string data with numeric dtype:

```python
>>> import pandas as pd
>>> pd.Index(["1", "2"], dtype="int64")
ValueError: invalid literal for int() with base 10: 's'  # different code path
```

Reply:
> so IMO a whatsnew notes should not be added

OK, but on the off-chance that the Int64Index removal doesn't happen in time, pls be sure to add a note.

Reply:
👍